src/Stream.ts

```diff
@@ -0,0 +1,108 @@
+import { Effect } from "./Effect";
+import * as Fx from "./Fx";
+import * as Task from "./Task";
+import * as Async from "./Async";
+import * as Fail from "./Fail";
+import * as Fork from "./Fork";
+
+export class Stream<A> extends Effect('Stream')<A, void> { }
+
+export namespace Stream { 
+  export type Event<T> = T extends Stream<infer A> ? A : never
+}
+
+export function event<const A>(a: A): Fx.Fx<Stream<A>, void> { 
+  return new Stream(a)
+}
+
+export function observe<E, R, E2>(fx: Fx.Fx<E, R>, f: (a: Stream.Event<E>) => Fx.Fx<E2, void>): Fx.Fx<Exclude<E, Stream<any>> | E2, R> { 
+  return fx.pipe(Fx.handle<typeof Stream, E2>(Stream, (a) => f(a as Stream.Event<E>)))
+} 
+
+export function collectAll<E, A>(fx: Fx.Fx<E, A>): Fx.Fx<Exclude<E, Stream<any>>, readonly Stream.Event<E>[]> {
```

Comment:
Yeah all good points, I really just hacked this for the sake of the tests I did write. 

TBH I haven't even actually utilized reduce in an application before. It assumes I want an ending state of single value and I feel that has mostly missed the mark on using a stream at that point. I mostly use it for the declarative syntax for concurrency patterns (mainly switchMap, debounce, throttle), various filtering, mapping, slicing, and for multicasting updates to multiple observers. I've definitely had a few more use-cases for scan and Effect variants of scan, but definitely not much.

I could honestly just see this function living in the test file, at least for now, what do you think?

Reply:
> TBH I haven't even actually utilized reduce in an application before

It probably depends on the use case.  In reactive use-cases, yeah, reduce usually isn't very useful.  In data processing, it can be very useful.  That said, I'm fine with not adding `reduce` right now.  map+filter+forEach is a very solid starting point!

> I could honestly just see this function living in the test file, at least for now, what do you think?

That sounds good to me for now 👍